Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -343,6 +345,59 @@ public static StringCollectionAssertions Should(this IEnumerable<string> @this)
             return new GenericDictionaryAssertions<TCollection, TKey, TValue>(actualValue);
         }
 
+        /// <summary>
+        /// Returns a <see cref="DataSetAssertions{DataSet}"/> object that can be used to assert the
+        /// current <see cref="DataSet"/>. To perform assertions based on a specialized compile-time
+        /// type, use the <see cref="DataSetAssertions{DataSet}.As"/> method.
+        /// </summary>
+        [Pure]
+        public static DataSetAssertions<DataSet> Should(this DataSet actualValue)
+        {
+            return new DataSetAssertions<DataSet>(actualValue);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="DataTableAssertions{DataTable}"/> object that can be used to assert the
+        /// current <see cref="DataTable"/>. To perform assertions based on a specialized compile-time
+        /// type, use the <see cref="DataTableAssertions{DataTable}.As"/> method.
+        /// </summary>
+        [Pure]
+        public static DataTableAssertions<DataTable> Should(this DataTable actualValue)
+        {
+            return new DataTableAssertions<DataTable>(actualValue);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="DataTableAssertions{TypedTableBase}"/> object that can be used to assert the
+        /// current typed <see cref="DataTable"/>.
+        /// </summary>
+        public static DataTableAssertions<TypedTableBase<T>> Should<T>(this TypedTableBase<T> actualValue)
+                where T : DataRow
+        {
+            return new DataTableAssertions<TypedTableBase<T>>(actualValue);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="DataColumnAssertions"/> object that can be used to assert the
+        /// current <see cref="DataColumn"/>.
+        /// </summary>
+        [Pure]
+        public static DataColumnAssertions Should(this DataColumn actualValue)
```

Comment:
Let's just make `DataColumnAssertions` generic as well while we're at it.

Reply:
There will never, ever be a subclass of `DataColumn` as far as I know.

Reply:
Okay, then fine to keep it as it is 👍